enhancements/rhcos/rhcos-cloud-marketplace-images.md

```diff
@@ -0,0 +1,330 @@
+* Definition in RHCOS stream to separate multiple similar offers (e.g. ROSA-specific marketplace images vs.
+OpenShift AWS marketplace images, Azure paid vs. free images)
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+This enhancement proposes a process for publishing the images and
+a specification for how the images would be published in the stream.
+
+The process for uploading images will be based on RHCOS stream bump
+PRs. New PRs to the installer will trigger a request to upload images.
+When the images are available, the PR will be merged to update the stream.
+
+Marketplace images will be included in the
+`.architectures[("aarch64","x86_64")].rhel_coreos_extensions` fields. 
+
+
+### Workflow Description
+
+The workflow for uploading cloud marketplace images is a
+pull-request-based workflow; the uploading of marketplace
+images will be initiated by a pull request bumping the RHCOS
+stream in the installer.
+
+**pull request** is a pull request, opened by an RHCOS engineer,
+bumping the RHCOS stream in openshift/installer. The bump
+is based 
+
+**software production** is the team of engineers who perform
+the marketplace image uploads.
+
+**cloud marketplaces** are AWS, Azure & GCP.
+
+```mermaid
+sequenceDiagram
+  participant PR as Pull Request
+  actor SP as Software Production
+  participant CM as Cloud Marketplaces
+
+  PR->>SP: When the pull request is opened, a JIRA ticket is generated and assigned to Software Production.
+  PR->>PR: E2E tests fail as marketplace images are not found
+  SP->>CM: Upload RHCOS images to cloud marketplaces
+  CM->>SP: Images are available
+  SP->>PR: JIRA ticket is closed
+  PR->>PR: Tests Pass
+```
+
+### API Extensions
+
+Machine pools in the install config would be updated to allow users to more simply
+opt in to using marketplace images without specifying particular image details.
+
+#### Azure
+
+Currently in Azure, to use paid marketplace images, [users consult
+documentation to determine image details](https://docs.openshift.com/container-platform/4.16/installing/installing_azure/installing-azure-customizations.html#installation-azure-marketplace-subscribe_installing-azure-customizations)
+and enter image details in the install config like this:
+
+```yaml
+compute:
+- name: worker
+  platform:
+    azure:
+      osImage:
+        publisher: RedHat
+        offer: rh-ocp-worker
+        sku: rh-ocp-worker
+        version: 413.92.2023101700
+```
+
+A `marketplaceImage` field would be added (OPEN QUESTION: either as a child of `osImage` or as a sibling)
+with the following enum values:
+
+* `NoPurchasePlan` - equivalent to a standard Azure OpenShift install, except under the
+hood we switch the use of the managed image with a non-paid OpenShift image
+* `PurchasePlanNorthAmerica` - equivalent to the yaml snippet directly preceding this
+* `PurchasePlanEMEA` - for users in EMEA, selects marketplace images from the `redhat-limited` publisher
```

Comment:
The last two here are hypothetical or sure things? Do we have any idea what sort of sku structure will be used or will it potentially change such that this should just be an opaque field, if it's empty use the baked in value which I'd assume is equivalent to `NoPurchasePlan` or use a given ARN/cloud reference string?

Reply:
The last two refer to the already published paid Azure marketplace images. `NoPurchasePlan` refers to the hypothetical non-paid image I would like to see published. I included output in the enhancement that shows the already published images (not sure how to link to it). The EMEA plan distinguishes itself by using a different publisher, `redhat-limited`.